python/dgl/data/citation_graph.py

```diff
@@ -29,34 +36,38 @@ def _pickle_load(pkl_file):
     else:
         return pkl.load(pkl_file)
 
-class CitationGraphDataset(object):
+class CitationGraphDataset(DGLBuiltinDataset):
     r"""The citation graph dataset, including cora, citeseer and pubmeb.
     Nodes mean authors and edges mean citation relationships.
 
     Parameters
     -----------
     name: str
       name can be 'cora', 'citeseer' or 'pubmed'.
+    raw_dir : str
+        Raw file directory to download/contains the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+      Whether to print out progress information. Default: True.
     """
-    def __init__(self, name):
+    def __init__(self, name, raw_dir=None, force_reload=False, verbose=True):
         assert name.lower() in ['cora', 'citeseer', 'pubmed']
```

Comment:
Can CoauthorDataset and AmazonCoBuyDataset in gnn_benchmark dataset use 'name' to identify different part of the dataset? 
For CoauthorDataset, there are two domains 'cs' and 'physics', now I split the dataset as CoauthorCSDataset and CoauthorPhysicsDataset.